File: sub-trafficparticipants/src/main/java/at/jku/trafficparticipants/entity/Street.java
package at.jku.trafficparticipants.entity;

public class Street extends Edge {

	private String name;
	
	private double length;
	
	private double currentTrafficRate;
	private double maxTrafficRate;
	private double defaultTrafficRate;
	
	private double currentSpeed;
	private double maxSpeed;
	private double defaultSpeed;
	
	private Node start;
	private Node end;
	
	public double getCurrentTrafficRate() {
		return currentTrafficRate;
	}
	
	public double getCurrentSpeed() {
		return currentSpeed;
	}
	
	public String getName() {
		return name;
	}
	public double getLength() {
		return length;
	}
	public double getMaxTrafficRate() {
		return maxTrafficRate;
	}
	public double getDefaultTrafficRate() {
		return defaultTrafficRate;
	}
	public double getMaxSpeed() {
		return maxSpeed;
	}
	public double getDefaultSpeed() {
		return defaultSpeed;
	}
	
	@Override
	public Node getStart() {
		return start;
	}
	
	@Override
	public Node getEnd() {
		return end;
	}
}

File: sub-controlsystem/src/main/java/at/jku/controlsystem/contract/Street.kt
package at.jku.controlsystem.contract

class Street(
        var name: String,
        var currentTrafficRate: Double,
        var defaultTrafficRate: Double,
        var currentSpeed: Double,
        var defaultSpeed: Double,
        override var startNode: Node?,
        override var endNode: Node?) : Edge{

    override fun hashCode(): Int {
        return name.hashCode() +
                currentTrafficRate.hashCode() +
                defaultTrafficRate.hashCode() +
                currentSpeed.hashCode() +
                defaultSpeed.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        return super.equals(other)
    }
}